eth-account-snapshot/index.js

```diff
@@ -1,70 +1,39 @@
 'use strict'
 const EthAccount = require('ethereumjs-account')
+const multicodec = require('multicodec')
+
 const cidFromHash = require('../util/cidFromHash')
 const createResolver = require('../util/createResolver')
 const emptyCodeHash = require('../util/emptyCodeHash')
 
-module.exports = createResolver('eth-account-snapshot', EthAccount, mapFromEthObj)
-
-
-function mapFromEthObj (account, options, callback) {
-  const paths = []
-
-  // external links
-
-  paths.push({
-    path: 'storage',
-    value: { '/': cidFromHash('eth-storage-trie', account.stateRoot).toBaseEncodedString() }
-  })
-
-  // resolve immediately if empty, otherwise link to code
-  if (emptyCodeHash.equals(account.codeHash)) {
-    paths.push({
-      path: 'code',
-      value: Buffer.from(''),
-    })
-  } else {
-    paths.push({
-      path: 'code',
-      value: { '/': cidFromHash('raw', account.codeHash).toBaseEncodedString() }
-    })
+const fieldAccess = () => {
+  const getters = {
+    storage: function () {
+      return cidFromHash(multicodec.ETH_STORAGE_TRIE, this.stateRoot)
+    },
+    code: function () {
+      // resolve immediately if empty, otherwise link to code
+      if (emptyCodeHash.equals(this.codeHash)) {
+        return Buffer.from('')
+      } else {
+        return cidFromHash(multicodec.RAW, this.codeHash)
+      }
+    }
   }
 
-  // external links as data
-
-  paths.push({
-    path: 'stateRoot',
-    value: account.stateRoot
-  })
-
-  paths.push({
-    path: 'codeHash',
-    value: account.codeHash
-  })
-
-  // internal data
-
-  paths.push({
-    path: 'nonce',
-    value: account.nonce
-  })
-
-  paths.push({
-    path: 'balance',
-    value: account.balance
-  })
-
-  // helpers
-
-  paths.push({
-    path: 'isEmpty',
-    value: account.isEmpty()
-  })
+  const values = {
+    isEmpty: function () { return this.isEmpty() },
```

Comment:
These don't look right, `this` is going to be the object they are attached to so `this.isEmpty()` should result in a recursive call. It's set as a 'value' but it's just a function, so I think you're replacing the existing `EthAccount#isEmpty()` with a new recursive one. What's the intent here?